.github/workflows/requirements.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: requirements
+
+on: [pull_request]
+
+jobs:
+  requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install covalent
+        run: python -m pip install .
+
+      - name: Install packages
+        run: python -m pip install pip-check-reqs
+
+      - name: Check extra core requirements
+        run: pip-extra-reqs -r werkzeug covalent covalent_dispatcher covalent_ui
+
+      - name: Check missing SDK requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
+          --ignore-module=urllib3.*
+          --ignore-module=pkg_resources
+          covalent
+
+      - name: Check missing dispatcher requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
+          --ignore-module=covalent_dispatcher
+          --ignore-module=distributed.*
+          --ignore-module=click
+          covalent_dispatcher
+
+      - name: Check missing UI backend requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
+          --ignore-module=covalent_dispatcher.*
+          --ignore-module=pydantic.*
```

Comment:
should `pydantic` be in the requirements?

Reply:
`pydantic` is included in `fastapi`

Reply:
```
  - fastapi [required: ==0.79.0, installed: 0.79.0]
    - pydantic [required: >=1.6.2,<2.0.0,!=1.8.1,!=1.8,!=1.7.3,!=1.7.2,!=1.7.1,!=1.7, installed: 1.10.1]
      - typing-extensions [required: >=4.1.0, installed: 4.3.0]
    - starlette [required: ==0.19.1, installed: 0.19.1]
      - anyio [required: >=3.4.0,<5, installed: 3.6.1]
        - idna [required: >=2.8, installed: 3.3]
        - sniffio [required: >=1.1, installed: 1.2.0]
```